server/blob/james-server-blob-api/src/main/java/org/apache/james/blob/api/ObjectStore.java

```diff
@@ -0,0 +1,28 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+package org.apache.james.blob.api;
+
+import java.util.concurrent.CompletableFuture;
+
+public interface ObjectStore {
+
+    CompletableFuture<BlobId> save(byte[] data);
```

Comment:
Did you read my comment about using {Input,Output}stream instead ?

Reply:
IMO we could have two methods types: 
 - One dealing with byte[], one for streams.

Reply:
what is the point, wrapping a byte[] in a stream is so easy

Reply:
Yes, but you might  loose the "completable future part"

Reply:
Yes, I have read your comment.
But you don't have read what I have said:
* in the PR description
* in our internal chat

I took two different media to explain that, I'm a bit surprised that you are asking this question.

Reply:
The downside of using a new PR.